rules/manager.go

```diff
@@ -317,7 +317,18 @@ func (g *Group) Name() string { return g.name }
 func (g *Group) File() string { return g.file }
 
 // Rules returns the group's rules.
-func (g *Group) Rules() []Rule { return g.rules }
+func (g *Group) Rules(matcherSets ...[]*labels.Matcher) []Rule {
+	if len(matcherSets) == 0 {
+		return g.rules
+	}
+	var rules []Rule
+	for _, rule := range g.rules {
+		if matches(matcherSets, rule.Labels()) {
```

Comment:
I guess we need special label matching for templates defined 🤔

Reply:
Or we keep it simple and update docs instead.

Reply:
Updated docs instead of matching templated labels. We instead check if the label values are templated. If they are templated they are skipped while matching, We do this by parsing the label value as a template and check for the number of nodes. If there is only one node and the type is [`parser.NodeText`](https://pkg.go.dev/text/template/parse@go1.17.6#NodeType) then it is treated as a non-templated label. Should this be a regex-based approach instead? cc: @roidelapluie